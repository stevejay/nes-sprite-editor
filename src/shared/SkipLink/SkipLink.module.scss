@import "styles/_base.scss";

.link {
  @include screen-reader-only;

  // from: https://gomakethings.com/hidden-content-for-better-a11y/#hiding-the-link
  &:active,
  &:focus {
    clip: auto;
    height: auto;
    margin: 0;
    overflow: visible;
    white-space: normal;
    width: auto;
    position: absolute;
    padding: rem(5) rem(8);
    border-radius: rem(6);
    background-color: rgba($md-white, 1);
  }
}
