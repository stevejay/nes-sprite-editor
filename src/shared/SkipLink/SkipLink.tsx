import React from "react";
import styles from "./SkipLink.module.scss";

type Props = {
  href: string;
  children: React.ReactNode;
};

// A useful a11y tool:
// https://gomakethings.com/hidden-content-for-better-a11y/#the-skip-link
const SkipLink = React.forwardRef<HTMLAnchorElement, Props>(
  ({ href, children }, ref) => (
    <a ref={ref} className={styles.link} href={href}>
      {children}
    </a>
  )
);

export default SkipLink;
