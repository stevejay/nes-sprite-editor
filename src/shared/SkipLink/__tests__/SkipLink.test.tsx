import React from "react";
import { render, cleanup } from "react-testing-library";
import SkipLink from "../SkipLink";

afterEach(cleanup);

test("displays correctly", async () => {
  const { container } = render(
    <SkipLink href="/some/href">The link text</SkipLink>
  );
  expect(container).toMatchSnapshot();
});
