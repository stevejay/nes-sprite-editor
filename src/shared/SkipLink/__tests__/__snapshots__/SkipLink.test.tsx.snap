// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correctly 1`] = `
<div>
  <a
    class="link"
    href="/some/href"
  >
    The link text
  </a>
</div>
`;
